lib/base/utility.cpp

```diff
@@ -1324,10 +1324,12 @@ void Utility::SetThreadName(const String& name, bool os)
 
 #ifdef HAVE_PTHREAD_SETNAME_NP
 #	ifdef __APPLE__
-	pthread_setname_np(name.CStr());
-#	else /* __APPLE__ */
-	String tname = name.SubStr(0, 15);
-	pthread_setname_np(pthread_self(), tname.CStr());
+		pthread_setname_np(name.CStr());
+#	elif defined(__NetBSD__) /* __APPLE__ */
+		pthread_setname_np(pthread_self(), "%s", const_cast<char *>(name.CStr()));
```

Comment:
In the [`pthread_setname_np(3)`](https://man.netbsd.org/pthread_setname_np.3) man page on NetBSD states that the resulting name should be shorter than `PTHREAD_MAX_NAMELEN_NP`, so I would also use the same substr used in the else case below here.

Reply:
Ya, I was going to submit another patch to adjust that for all of the cases as appropriate.

PTHREAD_MAX_NAMELEN_NP is 32. FreeBSD / DragonFly have a length of 16. OpenBSD has a length of 24. macOS has a length of 64.

Reply:
If you want to, just feel free to include it in a second commit in this PR.